pre-processing.py

```diff
@@ -0,0 +1,228 @@
+    person, demog_cols, w="asecwt"
+)
+# add name to series
+pov_rate_us.name = "US"
+# calculate poverty RATE for each group by state
+pov_rates = mdf.weighted_sum(
+    poor_pop, demog_cols, "asecwt", groupby="state"
+) / mdf.weighted_sum(person, demog_cols, w="asecwt", groupby="state")
+
+# append US statistics as additional 'state'
+pov_df = pov_rates.append(pov_rate_us)
+
+# melt df from wide to long format
+pov_df = pov_df.melt(ignore_index=False, var_name="demog")
+# insert column indicating metric in question
+pov_df.insert(loc=1, column="metric", value="pov_rate")
+
+
+##
+# calculate POPULATION for each DEMOGRAPHIC in US
+pop_us = mdf.weighted_sum(person, demog_cols, w="asecwt")
+# add name to series
+pop_us.name = "US"
+# calculate POPULATION for each group by state
+pop_states = mdf.weighted_sum(person, demog_cols, w="asecwt", groupby="state")
+# append US statistics as additional 'state'
+pop_df = pop_states.append(pop_us)
+# melt df from wide to long format
+pop_df = pop_df.melt(ignore_index=False, var_name="demog")
+pop_df.insert(loc=1, column="metric", value="pop")
+
+# concat poverty and population dfs
+demog_stats = pd.concat([pov_df, pop_df])
+# write to csv file
+demog_stats.to_csv("demog_stats.csv.gz", compression="gzip")
+
+# Caluclate original gini
+person["spm_resources_per_person"] = person.spmtotres / person.numper
+# Caluclate original gini for US
+gini_us = pd.Series(mdf.gini(df=person, col="spm_resources_per_person", w="asecwt"))
+# add name to series
+gini_us.index = ["US"]
+
+
+# calculate gini for each group by state
+gini_states = mdf.gini(
+    df=person, col="spm_resources_per_person", w="asecwt", groupby="state"
+)
+
+# append US statistics as additional 'state'
+gini_ser = gini_states.append(gini_us)
+gini_ser.name = "gini"
+
+# Calculate the original poverty gap
+spmu["poverty_gap"] = np.where(
+    spmu.spmtotres < spmu.spmthresh,
+    spmu.spmthresh - spmu.spmtotres,
+    0,
+)
+poverty_gap_us = pd.Series(mdf.weighted_sum(spmu, "poverty_gap", w="spmwt"))
+# add name to series
+poverty_gap_us.index = ["US"]
+# calculate gini for each group by state
+poverty_gap_states = mdf.weighted_sum(spmu, "poverty_gap", w="spmwt", groupby="state")
+# append US statistics as additional 'state'
+poverty_gap_ser = poverty_gap_states.append(poverty_gap_us)
+poverty_gap_ser.name = "poverty_gap"
+
+# calculate the sum total of everyone's resources in US
+total_resources_us = pd.Series(mdf.weighted_sum(spmu, "spmtotres", w="spmwt"))
+# add name to series
+total_resources_us.index = ["US"]
+# calculate gini for each group by state
+total_resources_state = mdf.weighted_sum(
+    df=spmu, col="spmtotres", w="spmwt", groupby="state"
+)
+# append US statistics as additional 'state'
+total_resources_state = total_resources_state.append(total_resources_us)
+total_resources_state.name = "total_resources"
+
+# merge "total_resources","gini","poverty gap" into 1 df
+all_state_stats = poverty_gap_ser.to_frame().merge(
```

Comment:
`DataFrame.join(DataFrame)` is equivalent to `DataFrame.merge(DataFrame, left_index=True, right_index=True)`.
```suggestion
all_state_stats = poverty_gap_ser.to_frame().join(total_resources_state.to_frame())
```